### PR TITLE
feat(surveys): Add question html support

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

Fixes https://github.com/PostHog/posthog/issues/17506

Depends on https://github.com/PostHog/posthog-js/pull/824

image

image

We sanitize inputs

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Review by jurajmajerik — APPROVED** [review]

**Review by liyiy — APPROVED** [review]

File: posthog/models/feature_flag/flag_matching.py [hunk 1]
```diff
@@ -496,6 +496,7 @@ def hashed_identifier(self, feature_flag: FeatureFlag) -> Optional[str]:
                     return self.hash_key_overrides[feature_flag.key]
             return self.distinct_id
         else:
+            # TODO: Don't use the cache if self.groups is empty, since that means no groups provided anyway
```
Comment (liyiy) [inline posthog/models/feature_flag/flag_matching.py:499]: is this a part of the PR? :o

**Comment by liyiy** [comment]
I'm going to test this locally and then merge if it's all good

**Comment by liyiy** [comment]
I feel like we should actually prevent the user from creating the survey if they have script in their description HTML. Removing them on save works too but since we're going to great lengths to prevent it anyway, maybe we should be more direct by just form erroring them entirely?

Anyway not really a blocker and everything else is working, so I'll leave it to you to decide!

**Comment by neilkakkar** [comment]
The tricky bit is that there are several ways to get a script inside:

```
DOMPurify.sanitize('<img src=x onerror=alert(1)//>'); // becomes <img src="x">
DOMPurify.sanitize('<svg><g/onload=alert(2)//<p>'); // becomes <svg><g></g></svg>
DOMPurify.sanitize('<p>abc<iframe//src=jAva&Tab;script:alert(3)>def</p>'); // becomes <p>abc</p>
DOMPurify.sanitize('<math><mi//xlink:href="data:x,<script>alert(4)</script>">'); // becomes <math><mi></mi></math>
DOMPurify.sanitize('<TABLE><tr><td>HELLO</tr></TABL>'); // becomes <table><tbody><tr><td>HELLO</td></tr></tbody></table>
DOMPurify.sanitize('<UL><li><A HREF=//google.com>click</UL>'); // becomes <ul><li><a href="//google.com">click</a></li></ul>
```

and there's a second benefit of sanitising which we seriously need: To fix malformed html, per user input error, or close unclosed tags. This doesn't affect the preview much, but would completely break the actual survey if there are malformed tags.

So, we're always doing some adjustments on save. And at this stage instead of making exceptions for a small class of script tags, I felt it better to just go the route of "we will clean up your HTML no matter what it is"

**Pull request merged**